### ISSUE TITLE
Code Workspace Classpath is corrupted after restart

### ISSUE THREAD

**Issue by clemenswasser** [redhat-developer/vscode-java#1941]
1. When I open a code-workspace with a clean Java Language Server Workspace everything works.
2. Once I restart VSCode and open any file, I get the following warning *SomeJavaClass*`.java is not on the classpath of project `*workspacename+some_random_hash*`, only syntax errors are reported`
3. When I execute `Java: Clean Java Language Server Workspace` we are back at 1.

##### Environment
- Operating System: Windows 10
- JDK version: 16.0.1.9
- Visual Studio Code version: 1.56.1
- Java extension version: 0.79.1

##### Steps To Reproduce
1. Create a `test.code-workspace` file with the following contents:
    ```
    {
        "folders": [
            {
                "path": "test"
            }
        ]
    }
    ```
2. Create folder `test`
3. Create a file `Test.java` in that folder
4. Open the code-workspace with VSCode
5. Open `test/Test.java` in VSCode
6. Everything works
7. Execute `Developer: Reload Window` or restart VSCode
8. Open `test/Test.java` in VSCode
9. **This Error appears and nothing works**
10. Execute `Java: Clean Java Language Server Workspace` and restart VSCode
11. Everything works

##### Current Result

A restart of VSCode breaks the Java Language Server Workspace and Classpath

##### Expected Result

It shouldn't

**Comment by rgrunber** [comment]
I thought this might be similar to https://github.com/eclipse/eclipse.jdt.ls/issues/1764, but I'm not able to reproduce this issue. @snjeza let me know if you have any luck here.

**Comment by clemenswasser** [comment]
> I thought this might be similar to https://github.com/eclipse/eclipse.jdt.ls/issues/1764, but I'm not able to reproduce this issue. @snjeza let me know if you have any luck here.

I remember also experiencing the issue you linked.

**Comment by snjeza** [comment]
@clemenswasser could you try https://github.com/snjeza/vscode-test/raw/master/java-0.79.3.vsix

**Comment by clemenswasser** [comment]
> I thought this might be similar to [eclipse/eclipse.jdt.ls#1764](https://github.com/eclipse/eclipse.jdt.ls/issues/1764), but I'm not able to reproduce this issue. @snjeza let me know if you have any luck here.

I have now reinstalled VSCode (including deleting configuration and cache) and am still able to reproduce this.
@snjeza Does this build that you posted contain a fix? Or why should I test it?

I have to add something:
I was previously using the `Java Extension Pack` and can still reproduce it using only `Language Support for Java(TM) by Red Hat` (`redhat.java`) and `Project Manager for Java` (vscjava.vscode-java-dependency).

**Comment by snjeza** [comment]
>  Does this build that you posted contain a fix?

@clemenswasser Yes, it includes https://github.com/eclipse/eclipse.jdt.ls/pull/1766

**Comment by snjeza** [comment]
> I was previously using the Java Extension Pack and can still reproduce it using only Language Support for Java(TM) by Red Hat (redhat.java) and Project Manager for Java (vscjava.vscode-java-dependency).

I can reproduce the issue with VS Code 0.79.1, but can't with https://github.com/snjeza/vscode-test/raw/master/java-0.79.3.vsix

**Comment by clemenswasser** [comment]
I have done some further investigations:
With the `redhat.java` extension standalone I cannot reproduce this.
Once I add any of the following extensions it does reproduce:
- `vscjava.vscode-java-test` (+ `vscjava.vscode-java-debug` this is a dependency of the Test Runner extension)
- `vscjava.vscode-maven`
- `vscjava.vscode-java-dependency`

It is really weird. I can only reproduce it when other extensions (which are part of the `Java Extension Pack`) are activated too.

**Comment by snjeza** [comment]
> It is really weird. I can only reproduce it when other extensions (which are part of the Java Extension Pack) are activated too.

@clemenswasser Did you try https://github.com/snjeza/vscode-test/raw/master/java-0.79.3.vsix

**Comment by clemenswasser** [comment]
> > It is really weird. I can only reproduce it when other extensions (which are part of the Java Extension Pack) are activated too.
> 
> @clemenswasser Did you try https://github.com/snjeza/vscode-test/raw/master/java-0.79.3.vsix

Sorry for not trying this directly. This solves the issue for me. Thank you very much 🥳
When will version 0.79.3 be released?

**Comment by snjeza** [comment]
@clemenswasser Could you try https://github.com/snjeza/vscode-test/raw/master/java-0.79.4.vsix ?

**Comment by clemenswasser** [comment]
> @clemenswasser Could you try https://github.com/snjeza/vscode-test/raw/master/java-0.79.4.vsix ?

This does also solve the issue for me.

**Comment by jdneo** [comment]
This should be fixed by https://github.com/eclipse/eclipse.jdt.ls/pull/1767 I guess. @snjeza @rgrunber

**Comment by rgrunber** [comment]
Yes, the change appears to resolve the issue. Cleaning the language server workspace, or reloading the window continue to work as well with the snippet provided.

This  will be fixed i 0.79.2, which we expect to have out early next week. Feel free to re-open the issue though if the problem does persist.
